src/hotspot/share/gc/serial/serialFullGC.cpp

```diff
@@ -604,11 +609,34 @@ template <class T> void SerialFullGC::mark_and_push(T* p) {
   }
 }
 
+inline size_t SerialFullGC::adjust_pointers(oop obj) {
+  return obj->oop_iterate_size(&SerialFullGC::adjust_pointer_closure);
+}
+
+template <class T> inline void SerialFullGC::adjust_pointer(T* p) {
```

Comment:
This new code should be after "void MarkAndPushClosure::do_oop(narrowOop* p) { do_oop_work(p); }" so that `mark_and_push` is next to its caller.

Reply:
Moved.

Reply:
The "inline" keyword can be dropped, both in hpp and cpp.

Reply:
Removed.